building-with-ocean/using-ocean-libraries/configuration.md

```diff
@@ -188,17 +188,26 @@ import os
 from dotenv import load_dotenv
 from ocean_lib.ocean.ocean import Ocean
 from ocean_lib.web3_internal.wallet import Wallet
-from ocean_lib.example_config import ExampleConfig, get_config_dict
+from ocean_lib.example_config import ExampleConfig
 from ocean_lib.ocean.ocean import Ocean
-from ocean_lib.ocean.util import get_web3
+from web3.middleware import geth_poa_middleware
 
 load_dotenv()
 
 config = ExampleConfig.get_config()
 ocean = Ocean(config)
 
-user_private_key = os.getenv('PRIVATE_KEY')
-web3_wallet = Wallet(ocean.web3, user_private_key, ocean.config.block_confirmations, ocean.config.transaction_timeout)
+user_private_key = os.getenv("PRIVATE_KEY")
+
+if config.network_name == "mumbai":
```

Comment:
this is not the appropriate fix. This should be based on a config, i.e like it is in provider (I think, it might be in Aquarius though), to use the USE_POA_MIDDLEWARE to determine injection. I think rinkeby also requires it. And the fix should be done in ocean.py itself, which provides templates for each network. Please create a ticket in ocean.py and I can handle it.

Reply:
ocean.py ticket is here: https://github.com/oceanprotocol/ocean.py/issues/891